app/src/lib/stores/repositories-store.ts

```diff
@@ -531,31 +557,100 @@ export class RepositoriesStore extends BaseStore {
     return record!.lastPruneDate
   }
 
-  public async isBranchProtected(
+  /**
+   * Inspect the cache or database to find out information about the branch
+   * protection configuration for the current repository and branch.
+   */
+  public async getBranchProtectionContext(
     gitHubRepository: GitHubRepository,
     branchName: string
-  ): Promise<boolean> {
+  ): Promise<BranchProtectionContext> {
     if (gitHubRepository.dbID === null) {
       return fatalError(
         'unable to get protected branches, GitHub repository has a null dbID'
       )
     }
 
-    const key = getKey(gitHubRepository.dbID, branchName)
+    const repoID = gitHubRepository.dbID
+
+    const branchProtectionsFound = this.branchProtectionSettingsFoundCache.get(
+      repoID
+    )
+
+    if (branchProtectionsFound === undefined) {
+      return this.loadAndCacheBranchProtection(repoID, branchName)
+    } else if (branchProtectionsFound) {
```

Comment:
this is redundant but made this easier for me to follow since it wasn't immediately apparent that `branchProtectionsFound` was a `boolean`. definitely non-blocking :)

```suggestion
    } else if (branchProtectionsFound === true) {
```